File: python/BinarySearch.py
class Solution:
    def bnrySrch(self, arr, x):
        low = 0
        high = len(arr) - 1
        mid = 0

        while low <= high:
            mid = (high + low) // 2

        # If x is greater, ignore left half
            if arr[mid] < x:
                low = mid + 1

        # If x is smaller, ignore right half
            elif arr[mid] > x:
                high = mid - 1

        # means x is present at mid
            else:
                return mid

    # If we reach here, then the element was not present
        return -1


nums = [1, 2, 24, 456, 8]
x = 456
sol = Solution()
val = sol.bnrySrch(nums, x)
print(val)


# nums = [1, 2, 24, 456, 8]
# print(nums[:5:3])

File: Java/BinarySearch.java
public class BinarySearch {


    public static int bnrySrch(int[] nums, int numberToFind)
    {
     
        int low = 0;
        int high = nums.length - 1;

        while (low <= high) {
            int middlePosition = (low + high) / 2;
            int middleNumber = nums[middlePosition];
            if (numberToFind == middleNumber) {
                return middlePosition;
            }
            if (numberToFind < middlePosition) {
                high = middlePosition - 1;
            }
            else {
                low = middlePosition + 1;   
            } 
        }
        return -1;
    }

    public static void main(String[] args) {
        int[] nums = { 1, 2, 3, 4, 5 };
        int val =bnrySrch(nums, 4);
        System.out.println(val);
    }
}

